### PR TITLE
Fix #469 by checking the value of `analysis.jmp.cref`

### LINKED ISSUE #469: `aar` ignores `analysis.jmp.cref` value
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | -
| File format of the file you reverse (mandatory)      | -
| Architecture/bits of the file (mandatory)            | -
| `rizin -v` full output, **not truncated** (mandatory)  | 6814610

Option `analysis.jmp.cref` is set to false by default but `aar` ignores its value and anyway creates xrefs for conditional jumps.

See `librz/core/cmd_analysis.c` for how `aar` command is implemented.

### PR DESCRIPTION
**Pull request by theopechli**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This issue was that during `aaa`, the config variable `analysis.jmp.cref` is ignored and, as such, `aar` is run despite its value.

Note that its default value is `false`, which may be undesirable.

**Test plan**

I can add tests tomorrow, if needed.

**Closing issues**

closes #469

### CODE REVIEW

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -6667,12 +6667,14 @@ RZ_API bool rz_core_analysis_everything(RzCore *core, bool experimental, char *d
 		}
 	}
 
-	oldstr = rz_print_rowlog(core->print, "Analyze len bytes of instructions for references (aar)");
-	(void)rz_core_analysis_refs(core, ""); // "aar"
-	rz_print_rowlog_done(core->print, oldstr);
-	rz_core_task_yield(&core->tasks);
-	if (rz_cons_is_breaked()) {
-		return false;
+	if (rz_config_get_b(core->config, "analysis.jmp.cref")) {
```
Comment (wargio) [inline librz/core/canalysis.c:6670]: Maybe i'm missing something here. why this flag?

**Comment by wargio** [comment]
not sure if this is the right way.. let's wait for the others and especially @pelijah

**Comment by p371jah** [comment]
Well I have no idea how it should be done.

**Comment by XVilka** [comment]
I checked some tests output with the `analysis.jmp.cref=true` by default and the output looks better, in fact. I am not sure it should be `false` by default. cc @wargio @thestr4ng3r @ret2libc

File: test/db/tools/rz_diff [hunk 2]
```diff
@@ -299,7 +299,7 @@ RUN
 
 NAME=rz-diff graphs comparison with addresses JSON
 FILE==
-CMDS=!rz-diff -C -j -t graphs -0 fcn.00401ae0 -1 fcn.00401af0 bins/other/rz-diff/true bins/other/rz-diff/false
+CMDS=!rz-diff -e analysis.jmp.cref=true -C -j -t graphs -0 fcn.00401ae0 -1 fcn.00401af0 bins/other/rz-diff/true bins/other/rz-diff/false
```
Comment (wargio) [inline test/db/tools/rz_diff:302]: there is no `-e` option on rz-diff

**Review by ret2libc — CHANGES_REQUESTED** [review]
I think @wargio is right. Your PR run or does not run `rz_core_analysis_refs` based on the `analysis.jmp.cref` variable. However, `analysis.jmp.cref` says `analysis.jmp.cref: Create references for conditional jumps`, while `rz_core_analysis_refs` does much more than just analyzing refs for conditional jumps, it actually defines refs for calls, unconditional branches, data, etc.

I think you should look at `rz_core_analysis_refs` to see what to modifies and in particular at `rz_core_analysis_search_xrefs`. Please let us know if you need additional help!

File: test/db/tools/rz_diff (hunk 2 above)
Comment (theopechli) [reply test/db/tools/rz_diff:302]: I've been fooled by `man rz-diff`

Comment (wargio) [reply test/db/tools/rz_diff:302]: yeah. that one is old

**Comment by theopechli** [comment]
The only test that's failing is `db/tools/rz_diff rz-diff graphs comparison with addresses JSON`, because `analysis.jmp.cref=false` by default and rz-diff has no option to configure variables, despite what `man rz-diff` prints.

Two obvious solutions are:
1. Set `analysis.jmp.cref` to true
2. Mark that test as broken

I don't know if I missed something else in this PR.

**Comment by wargio** [comment]
> The only test that's failing is `db/tools/rz_diff rz-diff graphs comparison with addresses JSON`, because `analysis.jmp.cref=false` by default and rz-diff has no option to configure variables, despite what `man rz-diff` prints.
> 
> Two obvious solutions are:
> 
> 1. Set `analysis.jmp.cref` to true
> 2. Mark that test as broken
> 
> I don't know if I missed something else in this PR.

i will try to add `-e` option to rz-diff today so you can rebase and it should work.

**Comment by ret2libc** [comment]
@wargio @theopechli what's the status of this? I think the patch looks good, why is it failing? Can we workaround the problem so that `-e` in rz-diff can be done at a later time?

**Comment by wargio** [comment]
no. i need to patch rz-diff and add the -e flag. i will try to add it tonight.

**Comment by ret2libc** [comment]
@wargio ping

**Comment by wargio** [comment]
i'm so sorry for the delay. i have added the option. once it is merged, we can update this

**Review by wargio — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Pull request merged**